beanie/odm/fields.py

```diff
@@ -540,6 +540,14 @@ def from_motor_index_information(cls, index_info: dict):
             if ("_id", 1) in fields:
                 continue
 
+            if (
+                fields
+                and isinstance(fields[0][1], float)
+                and fields[0][1] == 1.0
+                or fields[0][1] == -1.0
```

Comment:
This expression is equivalent (see [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence)) with
```python
(fields and isinstance(fields[0][1], float) and fields[0][1] == 1.0) or fields[0][1] == -1.0
```
So, when `fields` is evaluated as `False`, then `fields[0]` will raise an `IndexError` (or other error, if it is not a list) and that may not be what we want.

> Note: Even if `and` and `or` would have the same precedence, the same situation could be possible. That is why it is more readable to use parenthesis when different logical operators are present in an expression.

Reply:
This is called short circuiting and when fields is evaluated as False, it won't go ahead to check the further conditions

Reply:
The `and` sequence is not evaluated completely because of short circuiting, but the left side of `or` yes.
Try this out:
```python
0 and 1 and 2 or 3
```
The expression can be parenthesis-ed as `(0 and 1 and 2) or 3`

This conditional expression introduce a new bug.

If you would write test for the case when `bool(fields)` is Ḟalse˙, it would crash probably with `IndexError`.